flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/SocketClientSinkTest.java

```
@@ -0,0 +1,132 @@
+import java.net.ServerSocket;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Tests for the {@link org.apache.flink.streaming.api.functions.sink.SocketClientSink}.
+ */
+public class SocketClientSinkTest{
+
+	final AtomicReference<Throwable> error = new AtomicReference<Throwable>();
+	private final String host = "127.0.0.1";
+	private int port;
+	private String access;
+	private String value;
+	public SocketServer.ServerThread th;
+
+	public SocketClientSinkTest() {
+	}
+
+	class SocketServer extends Thread {
+
+		private ServerSocket server;
+		private Socket sk;
+		private BufferedReader rdr;
+
+		private SocketServer() {
+			try {
+				this.server = new ServerSocket(0);
+				port = server.getLocalPort();
+			} catch (Exception e) {
+				error.set(e);
+			}
+		}
+
+		public void run() {
+			try {
+				sk = server.accept();
+				access = "Connected";
+				th = new ServerThread(sk);
+				th.start();
+			} catch (Exception e) {
+				error.set(e);
+			}
+		}
+
+		class ServerThread extends Thread {
+			Socket sk;
+
+			public ServerThread(Socket sk) {
+				this.sk = sk;
+			}
+
+			public void run() {
+				try {
+					rdr = new BufferedReader(new InputStreamReader(sk
+							.getInputStream()));
+					value = rdr.readLine();
+				} catch (IOException e) {
+					error.set(e);
+				}
+			}
+		}
+	}
+
+	@Test
+	public void testSocketSink() throws Exception{
+
+		SocketServer server = new SocketServer();
+		server.start();
+
+		SerializationSchema<String, byte[]> simpleSchema = new SerializationSchema<String, byte[]>() {
+			@Override
+			public byte[] serialize(String element) {
+				return element.getBytes();
+			}
+		};
+
+		SocketClientSink<String> simpleSink = new SocketClientSink<String>(host, port, simpleSchema);
+		simpleSink.open(new Configuration());
+		simpleSink.invoke("testSocketSinkInvoke");
+		simpleSink.close();
+		try {
+			server.join();
+			th.join();
+		}
+		catch (Exception e){
+			Assert.fail(e.getMessage());
```

You don't need to catch the exception here, you can let it propagate out of the test.
It you choose to catch it, you should print it, so the log reveals problems on test failures.